src/apps/watchfaces/watchface_dual_time.cpp

```diff
@@ -0,0 +1,294 @@
+
+  ////date
+  hal->gfx()->setTextSize(2);
+  hal->gfx()->setTextMiddleAligned();
+  hal->gfx()->setTextLeftAligned();
+  hal->gfx()->setTextBottomAligned();
+  hal->gfx()->setTextCenterAligned();
+  hal->gfx()->setTextCursor(BUF_W / 2 / 2, 98);
+  {
+    char date[20];
+    // i really would want the date to be dynamic based on what's in the config, but the most efficient thing to do
+    // right now is simply three if statements covering the 3 common conditions.
+    switch (showDateFormat) {
+      case 1:  // 0 : mm/dd/yyyy
+        sprintf(date, "%02d/%02d/%02d", monthInt, dayInt, yearInt % 100);
+        break;
+      case 2:  // 1 : dd.mm.yyyy
+        sprintf(date, "%02d.%02d.%02d", dayInt, monthInt, yearInt % 100);
+        break;
+      case 3:  // 2 : yy.mm/dd
+        sprintf(date, "%02d.%02d/%02d", yearInt % 100, monthInt, dayInt);
+        break;
+    }
+    hal->gfx()->print(date);
+  }
+}
+void dualDateDisplay(const uint8_t& showDateFormat, uint8_t startCoord) {
+  uint32_t dayInt = 0;
+  uint32_t monthInt = 0;
+  uint32_t yearInt = 0;
+  OswHal* hal = OswHal::getInstance();
+  const char* weekday = hal->getDualWeekday();
+
+  hal->getDualDate(&dayInt, &monthInt, &yearInt);
+
+  // we want to output a value like "Wed, 05/02/2021"
+
+  // left dual Digital Watch
+  hal->gfx()->setTextSize(1);
+  hal->gfx()->setTextMiddleAligned();
+  hal->gfx()->setTextLeftAligned();
+  hal->gfx()->setTextCursor(BUF_W / 2 + startCoord, 75);
+
+  {
+    char weekday3[4];
+    weekday3[0] = weekday[0];
+    weekday3[1] = weekday[1];
+    weekday3[2] = weekday[2];
+    weekday3[3] = '\0';
+    hal->gfx()->print(weekday3);
+  }
+
+  // The GFX library has an alignment bug, causing single letters to "float", therefore the workaround above is used to
+  // still utilize the correct string printing.
+  // hal->gfx()->print(weekday[0]);
+  // hal->gfx()->print(weekday[1]);
+  // hal->gfx()->print(weekday[2]);
+
+  ////date
+  hal->gfx()->setTextSize(2);
+  hal->gfx()->setTextMiddleAligned();
+  hal->gfx()->setTextLeftAligned();
+  hal->gfx()->setTextBottomAligned();
+  hal->gfx()->setTextCenterAligned();
+  hal->gfx()->setTextCursor(BUF_W / 4 * 3, 98);
+  {
+    char date[20];
+    // i really would want the date to be dynamic based on what's in the config, but the most efficient thing to do
+    // right now is simply three if statements covering the 3 common conditions.
+    switch (showDateFormat) {
+      case 1:  // 0 : mm/dd/yyyy
+        sprintf(date, "%02d/%02d/%02d", monthInt, dayInt, yearInt % 100);
+        break;
+      case 2:  // 1 : dd.mm.yyyy
+        sprintf(date, "%02d.%02d.%02d", dayInt, monthInt, yearInt % 100);
+        break;
+      case 3:  // 2 : yy.mm/dd
+        sprintf(date, "%02d.%02d/%02d", yearInt % 100, monthInt, dayInt);
+        break;
+    }
+    hal->gfx()->print(date);
+  }
+}
+
+void timeDisplay(uint32_t hour, uint32_t minute, uint32_t second) {
+  OswHal* hal = OswHal::getInstance();
+  {
+    char time[20];
```

Comment:
Maybe just use the `String` class instead? It won't require you to allocate that in the stack. Just my personal opinion.

Reply:
The reason for sticking to this method is to accurately position the title(date/time/health-data). This allows you to find the exact median regardless of the length of the string. I'm looking for a `format` method other than this one, do you have any ideas?